includes/templates/class-amp-post-template.php

```diff
@@ -111,23 +111,24 @@ public function __construct( $post ) {
 		$this->data = [
```

Comment:
Instead of assembling all of this data in the constructor, I think it would be better to lazily construct it upon invoking the getter the first time. This would allow us to construct the `AMP_Post_Template` preemptively without potentially it ever being used (and thus avoiding wasted queries), such as when normal WordPress templates are being used in the `amp` theme directory.

We could then still pass the `AMP_Post_Template` object in the existing hooks like `amp_post_template_head`, and as long as no hooks call the getter, then the data is not fetched.

Reply:
A "fun" way of doing that (which doesn't require conditionals all over the place) is to not define the `$data` property at all in the class and then use `__get()` to add it the first time it is being requested (through whatever means). After that, it is just present, so the `__get()` won't be hit a second time.

Reply:
Yes, this is what I intended to say. It's just in this class that logic would be in `\AMP_Post_Template::get()`.

Reply:
Done in c74ba974